### PR TITLE
chore: update go linting + update dependencies for controller

### PR DESCRIPTION
**Pull request by thesuperzapper**
This PR makes the following changes:

1. Updates to dependencies used on the controller:

      - Go 1.21 -> 1.22
      - controller-runtime: 0.17.3 -> 0.19.1
      - golangci-lint: 1.57.2 -> 1.61.0
      - EnvTest: 1.29 -> 1.31

2. Updates to dependencies used by backend:

      - controller-runtime: 0.17.3 -> 0.19.1
      - golangci-lint: 1.57.2 -> 1.61.0
      - EnvTest: 1.29 -> 1.31

3. Improved the linting for all go projects (controller, backend):

     - We now enforce comment formats
     - We now enforce license headers
     - We now do lots of code smell checks
     - We now have a spell checker

### CODE REVIEW

**Comment by thesuperzapper** [comment]
@ederign can you please LGTM when you are ready, this is an important update, and we need to get other PRs to rebase on it so they get the new linting.

/approve

File: workspaces/controller/internal/webhook/workspacekind_webhook_test.go [hunk 1]
```diff
@@ -96,7 +96,6 @@ var _ = Describe("WorkspaceKind Webhook", func() {
 		}
 
 		for _, tc := range testCases {
-			tc := tc // Create a new instance of tc to avoid capturing the loop variable.
```
Comment (ederign) [inline workspaces/controller/internal/webhook/workspacekind_webhook_test.go:99]: @thesuperzapper I want to double-check with you that this doesn't break anything. Based on the comment, the variable looks pretty intentional :)

File: workspaces/controller/config/rbac/role.yaml [hunk 2]
```diff
@@ -4,37 +4,11 @@ kind: ClusterRole
 metadata:
   name: manager-role
 rules:
-- apiGroups:
```
Comment (ederign) [inline workspaces/controller/config/rbac/role.yaml:7]: Also, are sure about changing the roles?

**Review by ederign — CHANGES_REQUESTED** [review]
@thesuperzapper thank you so much for this important PR. Just want to double check a few things before +1.

**Review by ederign — APPROVED** [review]

**Pull request merged**